pkg/spdk/replica.go

```diff
@@ -1285,75 +1336,148 @@ func (r *Replica) RebuildingDstStart(spdkClient *spdkclient.Client, exposeRequir
 	}()
 
 	if r.State != types.InstanceStateRunning {
-		return "", fmt.Errorf("invalid state %v for replica %s rebuilding start", r.State, r.Name)
+		return "", fmt.Errorf("invalid state %v for dst replica %s rebuilding start", r.State, r.Name)
 	}
-	if r.isRebuilding || r.rebuildingLvol != nil {
+	if r.isRebuilding {
 		return "", fmt.Errorf("replica %s rebuilding is in process", r.Name)
 	}
 
 	defer func() {
 		if err != nil {
 			if r.State != types.InstanceStateError {
 				r.State = types.InstanceStateError
-				updateRequired = true
 			}
 			r.ErrorMsg = err.Error()
+			if r.rebuildingDstCache.rebuildingError == "" {
+				r.rebuildingDstCache.rebuildingError = err.Error()
+			}
 		} else {
 			if r.State != types.InstanceStateError {
 				r.ErrorMsg = ""
 			}
 		}
+		if r.rebuildingDstCache.rebuildingError != "" && r.rebuildingDstCache.rebuildingState != types.ProgressStateError {
+			r.rebuildingDstCache.rebuildingState = types.ProgressStateError
+		}
```

Comment:
Can we directly update `r.rebuildingDstCache.rebuildingState` to `types.ProgressStateError` while updating `rebuildingError`?